tests/JobDefinitionFactoryTest.php

```diff
@@ -0,0 +1,193 @@
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+                'backend' => [
+                    'type' => 'invalid',
+                ],
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+            'backend' => [
+                'type' => 'custom',
+            ],
+        ];
+
+        $jobDefinitions = $this->createJobDefinitionsWithConfigData($jobData, $configData);
+        self::assertSame(
+            'custom',
+            $jobDefinitions[0]->getConfiguration()['runtime']['backend']['type'] ?? null
+        );
+    }
+
+    public function testCreateJobDefinitionWithConfigId(): void
+    {
+        $configuration = [
+            'id' => 'my-config',
+            'version' => '1',
+            'state' => [],
+            'rows' => [],
+            'configuration' => [
+                'runtime' => [
+                    'foo' => 'bar',
+                ],
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+        ];
+
+        $jobDefinitions = $this->createJobDefinitionsWithConfiguration($jobData, $configuration);
+
+        self::assertCount(1, $jobDefinitions);
+        $jobDefinition = $jobDefinitions[0];
+
+        self::assertSame($jobData['configId'], $jobDefinition->getConfigId());
+        self::assertSame($jobData['componentId'], $jobDefinition->getComponentId());
+        self::assertSame('bar', $jobDefinition->getConfiguration()['runtime']['foo'] ?? null);
+    }
+
+    public function testCreateJobDefinitionWithConfigIdAndBackend(): void
+    {
+        $configuration = [
+            'id' => 'my-config',
+            'version' => '1',
+            'state' => [],
+            'rows' => [],
+            'runtime' => [
+                'foo' => 'bar',
+                'backend' => [
+                    'type' => 'invalid',
+                ],
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'backend' => [
+                'type' => 'custom',
+            ],
+        ];
+
+        $jobDefinitions = $this->createJobDefinitionsWithConfiguration($jobData, $configuration);
+
+        self::assertSame(
+            'custom',
+            $jobDefinitions[0]->getConfiguration()['runtime']['backend']['type'] ?? null
+        );
+    }
+
+    private function createJobDefinitionsWithConfigData(array $jobData, array $configData): array
```

Comment:
no a teď jsme došli k tomu, že configData se tu vůbec na nic nepoužívá :)

Reply:
jak to? vraci se z toho mocku encryptoru

Reply:
Jasně, ale to tím jen tak proteče